src/OrleansRuntime/Silo/Silo.cs

```diff
@@ -307,6 +318,7 @@ internal Silo(string name, SiloType siloType, ClusterConfiguration config, ILoca
             incomingAgent = new IncomingMessageAgent(Message.Categories.Application, messageCenter, activationDirectory, scheduler, dispatcher);
 
             membershipFactory = new MembershipFactory();
+            multiClusterFactory = new MultiClusterOracleFactory();
```

Comment:
not necessarily. Seems o.k. to me to have the one condition in `MultiClusterOracleFactory.CreateGossipOracle` than having conditions in multiple places.